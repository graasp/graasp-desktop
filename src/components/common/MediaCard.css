.MediaCardDescriptionText p {
  font-size: large;
}
