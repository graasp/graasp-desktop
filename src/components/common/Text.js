import React from 'react';
import PropTypes from 'prop-types';
import ReactQuill from 'react-quill';
import { hasMath, renderMath } from '../../utils/math';

const modules = {
  toolbar: false,
};

const formats = [
  'header',
  'bold',
  'italic',
  'underline',
  'strike',
  'blockquote',
  'list',
  'bullet',
  'indent',
  'link',
  'image',
  'formula',
];

const Text = ({ content, style, className }) => {
  let parsedContent = content;
  if (hasMath(content)) {
    parsedContent = renderMath(parsedContent);
  }
  return (
    <div style={style}>
      <ReactQuill
        className={className}
        value={parsedContent}
        modules={modules}
        formats={formats}
        style={{ border: '0' }}
        readOnly
      />
    </div>
  );
};

Text.propTypes = {
  content: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.shape({}),
};

Text.defaultProps = {
  content: '',
  className: '',
  style: {},
};

export default Text;
