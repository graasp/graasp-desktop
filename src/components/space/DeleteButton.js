import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DeleteIcon from '@material-ui/icons/Delete';
import { withTranslation } from 'react-i18next';
import IconButton from '@material-ui/core/IconButton/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import { withStyles } from '@material-ui/core';
import Styles from '../../Styles';
import { deleteSpace } from '../../actions/space';

class DeleteButton extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    classes: PropTypes.shape({ appBar: PropTypes.string.isRequired })
      .isRequired,
    dispatchDeleteSpace: PropTypes.func.isRequired,
    t: PropTypes.func.isRequired,
  };

  handleDelete = () => {
    const { id, dispatchDeleteSpace } = this.props;
    dispatchDeleteSpace({ id });
  };

  render() {
    const { classes, t } = this.props;
    return (
      <Tooltip title={t('Delete this space.')}>
        <IconButton
          color="inherit"
          onClick={this.handleDelete}
          className={classes.button}
        >
          <DeleteIcon />
        </IconButton>
      </Tooltip>
    );
  }
}

const mapDispatchToProps = {
  dispatchDeleteSpace: deleteSpace,
};

const ConnectedComponent = connect(
  null,
  mapDispatchToProps
)(DeleteButton);

const StyledComponent = withStyles(Styles, { withTheme: true })(
  ConnectedComponent
);

const TranslatedComponent = withTranslation()(StyledComponent);

export default TranslatedComponent;
