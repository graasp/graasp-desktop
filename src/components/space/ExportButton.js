import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import IconButton from '@material-ui/core/IconButton/IconButton';
import UnarchiveIcon from '@material-ui/icons/Unarchive';
import Tooltip from '@material-ui/core/Tooltip';
import { withStyles } from '@material-ui/core';
import Styles from '../../Styles';
import { exportSpace } from '../../actions/space';

class ExportButton extends Component {
  static propTypes = {
    space: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    classes: PropTypes.shape({ appBar: PropTypes.string.isRequired })
      .isRequired,
    dispatchExportSpace: PropTypes.func.isRequired,
    t: PropTypes.func.isRequired,
  };

  handleExport = () => {
    const { space, dispatchExportSpace } = this.props;
    const { id, name } = space;
    dispatchExportSpace(id, name);
  };

  render() {
    const { classes, t } = this.props;
    return (
      <Tooltip title={t('Export this space to share with others.')}>
        <IconButton
          color="inherit"
          onClick={this.handleExport}
          className={classes.button}
        >
          <UnarchiveIcon />
        </IconButton>
      </Tooltip>
    );
  }
}
const mapDispatchToProps = {
  dispatchExportSpace: exportSpace,
};

const ConnectedComponent = connect(
  null,
  mapDispatchToProps
)(ExportButton);

const StyledComponent = withStyles(Styles, { withTheme: true })(
  ConnectedComponent
);

const TranslatedComponent = withTranslation()(StyledComponent);

export default TranslatedComponent;
