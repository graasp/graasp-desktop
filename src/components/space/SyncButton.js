import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withTranslation } from 'react-i18next';
import IconButton from '@material-ui/core/IconButton/IconButton';
import SyncIcon from '@material-ui/icons/Sync';
import { Online } from 'react-detect-offline';
import Tooltip from '@material-ui/core/Tooltip';
import { withStyles } from '@material-ui/core';
import Styles from '../../Styles';
import { syncSpace } from '../../actions/space';

class SyncButton extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    classes: PropTypes.shape({ appBar: PropTypes.string.isRequired })
      .isRequired,
    dispatchSyncSpace: PropTypes.func.isRequired,
    t: PropTypes.func.isRequired,
  };

  handleSync = () => {
    const { id, dispatchSyncSpace } = this.props;
    dispatchSyncSpace({ id });
  };

  render() {
    const { classes, t } = this.props;
    return (
      <Online>
        <Tooltip
          title={t(
            'Synchronize this space with its online version. All user input will be deleted.'
          )}
        >
          <IconButton
            color="inherit"
            className={classes.button}
            onClick={this.handleSync}
          >
            <SyncIcon className={classes.rightIcon} />
          </IconButton>
        </Tooltip>
      </Online>
    );
  }
}
const mapDispatchToProps = {
  dispatchSyncSpace: syncSpace,
};

const ConnectedComponent = connect(
  null,
  mapDispatchToProps
)(SyncButton);

const StyledComponent = withStyles(Styles, { withTheme: true })(
  ConnectedComponent
);

const TranslatedComponent = withTranslation()(StyledComponent);

export default TranslatedComponent;
