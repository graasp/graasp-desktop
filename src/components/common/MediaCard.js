import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import DeleteButton from '../space/DeleteButton';
import ExportButton from '../space/ExportButton';
import SyncButton from '../space/SyncButton';
import { MIN_CARD_WIDTH } from '../../config/constants';

const styles = theme => ({
  card: {
    width: '100%',
    minWidth: MIN_CARD_WIDTH,
    margin: 'auto',
    marginBottom: 15,
  },
  cardDescription: { margin: 0, paddingTop: 0, paddingBottom: 0 },
  media: {
    height: 300,
  },
  leftIcon: {
    marginRight: theme.spacing(),
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
});

const MediaCard = props => {
  const { classes, image, text, viewLink, space } = props;
  const { id, name } = space;
  const [expanded, setExpanded] = React.useState(false);
  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.card}>
      <CardActionArea onClick={viewLink}>
        <CardMedia className={classes.media} image={image} title={name} />

        <CardContent>
          <Typography variant="h5" component="h2">
            {name}
          </Typography>
        </CardContent>
      </CardActionArea>

      <Collapse disableSpacing in={expanded} timeout="auto" unmountOnExit>
        <CardContent className={classes.cardDescription}>
          <Typography
            component="p"
            dangerouslySetInnerHTML={{ __html: text }}
          />
        </CardContent>
      </Collapse>

      <CardActions disableSpacing>
        <DeleteButton id={id} />
        <ExportButton space={space} />
        <SyncButton id={id} />

        {text && (
          <IconButton
            className={clsx(classes.expand, {
              [classes.expandOpen]: expanded,
            })}
            onClick={handleExpandClick}
            aria-expanded={expanded}
            aria-label="Show more"
          >
            <ExpandMoreIcon />
          </IconButton>
        )}
      </CardActions>
    </Card>
  );
};

MediaCard.propTypes = {
  classes: PropTypes.shape({ media: PropTypes.string.isRequired }).isRequired,
  space: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  image: PropTypes.string.isRequired,
  text: PropTypes.string,
  viewLink: PropTypes.func.isRequired,
};

MediaCard.defaultProps = {
  text: '',
};

export default withStyles(styles)(MediaCard);
