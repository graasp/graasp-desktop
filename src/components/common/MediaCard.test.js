import React from 'react';
import { shallow } from 'enzyme';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { MediaCard } from './MediaCard';
import DeleteButton from '../space/DeleteButton';
import ExportButton from '../space/ExportButton';
import SyncButton from '../space/SyncButton';
import Text from './Text';

const createMediaCardProps = (showActions, text) => {
  return {
    space: {
      id: 'id',
      name: 'name',
    },
    image: 'image',
    text,
    viewLink: jest.fn(),
    showActions,
    classes: {
      card: '',
      cardDescription: '',
      cardDescriptionText: '',
      media: '',
      leftIcon: '',
      expand: '',
      expandOpen: '',
    },
  };
};

describe('<MediaCard />', () => {
  // use defaultProps = false
  describe('<MediaCard /> with showActions = undefined', () => {
    let wrapper;
    let props;

    beforeAll(() => {
      props = createMediaCardProps(undefined, 'text');
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MediaCard {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <List /> component', () => {
      expect(wrapper.find(Card)).toHaveLength(1);
    });

    it('renders one <CardActionArea /> component with CardMedia and Typography', () => {
      const { space, image } = props;

      const cardActionArea = wrapper.find(CardActionArea);
      expect(cardActionArea).toHaveLength(1);

      const cardMedia = cardActionArea.find(CardMedia);
      expect(cardMedia).toHaveLength(1);
      expect(cardMedia.prop('image')).toEqual(image);
      expect(cardMedia.prop('title')).toEqual(space.name);

      const cardContent = cardActionArea.find(CardContent);
      expect(cardContent).toHaveLength(1);

      const typography = cardContent.find(Typography);
      expect(typography).toHaveLength(1);
      expect(typography.contains(space.name));
    });

    it('renders one <Collapse /> component with space description', () => {
      const collapse = wrapper.find(Collapse);
      expect(collapse).toHaveLength(1);
      // not expanded by default
      expect(collapse.prop('in')).toEqual(false);

      const cardContent = collapse.find(CardContent);
      expect(cardContent).toHaveLength(1);

      const text = cardContent.find(Text);
      expect(text).toHaveLength(1);
      expect(text.prop('content')).toEqual(props.text);
    });

    it('does not render <CardActions /> component', () => {
      expect(wrapper.find(CardActions)).toHaveLength(0);
    });
  });

  describe('<MediaCard /> with showActions = false', () => {
    let wrapper;
    let props;

    beforeAll(() => {
      props = createMediaCardProps(false, 'text');
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MediaCard {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders one <List /> component', () => {
      expect(wrapper.find(Card)).toHaveLength(1);
    });

    it('renders one <CardActionArea /> component with CardMedia and Typography', () => {
      const { space, image } = props;

      const cardActionArea = wrapper.find(CardActionArea);
      expect(cardActionArea).toHaveLength(1);

      const cardMedia = cardActionArea.find(CardMedia);
      expect(cardMedia).toHaveLength(1);
      expect(cardMedia.prop('image')).toEqual(image);
      expect(cardMedia.prop('title')).toEqual(space.name);

      const cardContent = cardActionArea.find(CardContent);
      expect(cardContent).toHaveLength(1);

      const typography = cardContent.find(Typography);
      expect(typography).toHaveLength(1);
      expect(typography.contains(space.name));
    });

    it('renders one <Collapse /> component with space description', () => {
      const collapse = wrapper.find(Collapse);
      expect(collapse).toHaveLength(1);
      // not expanded by default
      expect(collapse.prop('in')).toEqual(false);

      const cardContent = collapse.find(CardContent);
      expect(cardContent).toHaveLength(1);

      const text = cardContent.find(Text);
      expect(text).toHaveLength(1);
      expect(text.prop('content')).toEqual(props.text);
    });

    it('does not render <CardActions /> component', () => {
      expect(wrapper.find(CardActions)).toHaveLength(0);
    });
  });

  describe('<MediaCard /> with showActions = true', () => {
    describe('with text defined', () => {
      let wrapper;
      let props;

      beforeAll(() => {
        props = createMediaCardProps(true, 'text');
        // eslint-disable-next-line react/jsx-props-no-spreading
        wrapper = shallow(<MediaCard {...props} />);
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('renders one <List /> component', () => {
        expect(wrapper.find(Card)).toHaveLength(1);
      });

      it('renders one <CardActionArea /> component with CardMedia and Typography', () => {
        const { space, image } = props;

        const cardActionArea = wrapper.find(CardActionArea);
        expect(cardActionArea).toHaveLength(1);

        const cardMedia = cardActionArea.find(CardMedia);
        expect(cardMedia).toHaveLength(1);
        expect(cardMedia.prop('image')).toEqual(image);
        expect(cardMedia.prop('title')).toEqual(space.name);

        const cardContent = cardActionArea.find(CardContent);
        expect(cardContent).toHaveLength(1);

        const typography = cardContent.find(Typography);
        expect(typography).toHaveLength(1);
        expect(typography.contains(space.name));
      });

      it('renders one <Collapse /> component with space description', () => {
        const collapse = wrapper.find(Collapse);
        expect(collapse).toHaveLength(1);

        // not expanded by default
        expect(collapse.prop('in')).toEqual(false);

        const cardContent = collapse.find(CardContent);
        expect(cardContent).toHaveLength(1);

        const text = cardContent.find(Text);
        expect(text).toHaveLength(1);
        expect(text.prop('content')).toEqual(props.text);
      });

      it('renders <CardActions /> component with Buttons', () => {
        const { space } = props;

        const cardActions = wrapper.find(CardActions);
        expect(cardActions).toHaveLength(1);

        const deleteButton = cardActions.find(DeleteButton);
        expect(deleteButton).toHaveLength(1);
        expect(deleteButton.prop('id')).toEqual(space.id);

        const exportButton = cardActions.find(ExportButton);
        expect(exportButton).toHaveLength(1);
        expect(exportButton.prop('space')).toEqual(space);

        const syncButton = cardActions.find(SyncButton);
        expect(syncButton).toHaveLength(1);
        expect(syncButton.prop('id')).toEqual(space.id);
      });

      it('renders <IconButton /> component with <ExpandMoreIcon />', () => {
        const iconButton = wrapper.find(IconButton);
        expect(iconButton).toHaveLength(1);
        expect(iconButton.prop('aria-expanded')).toEqual(false);

        const expandMoreIcon = iconButton.find(ExpandMoreIcon);
        expect(expandMoreIcon).toHaveLength(1);
      });
    });

    describe('with text undefined', () => {
      let wrapper;
      let props;

      beforeAll(() => {
        props = createMediaCardProps(true, undefined);
        // eslint-disable-next-line react/jsx-props-no-spreading
        wrapper = shallow(<MediaCard {...props} />);
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('renders one <List /> component', () => {
        expect(wrapper.find(Card)).toHaveLength(1);
      });

      it('renders one <CardActionArea /> component with CardMedia and Typography', () => {
        const { space, image } = props;

        const cardActionArea = wrapper.find(CardActionArea);
        expect(cardActionArea).toHaveLength(1);

        const cardMedia = cardActionArea.find(CardMedia);
        expect(cardMedia).toHaveLength(1);
        expect(cardMedia.prop('image')).toEqual(image);
        expect(cardMedia.prop('title')).toEqual(space.name);

        const cardContent = cardActionArea.find(CardContent);
        expect(cardContent).toHaveLength(1);

        const typography = cardContent.find(Typography);
        expect(typography).toHaveLength(1);
        expect(typography.contains(space.name));
      });

      it('renders one <Collapse /> component with space description', () => {
        const collapse = wrapper.find(Collapse);
        expect(collapse).toHaveLength(1);
        // not expanded by default
        expect(collapse.prop('in')).toEqual(false);

        const cardContent = collapse.find(CardContent);
        expect(cardContent).toHaveLength(1);

        const text = cardContent.find(Text);
        expect(text).toHaveLength(1);

        // show default text
        expect(text.prop('content')).toEqual(MediaCard.defaultProps.text);
      });

      it('renders <CardActions /> component with Buttons', () => {
        const { space } = props;

        const cardActions = wrapper.find(CardActions);
        expect(cardActions).toHaveLength(1);

        const deleteButton = cardActions.find(DeleteButton);
        expect(deleteButton).toHaveLength(1);
        expect(deleteButton.prop('id')).toEqual(space.id);

        const exportButton = cardActions.find(ExportButton);
        expect(exportButton).toHaveLength(1);
        expect(exportButton.prop('space')).toEqual(space);

        const syncButton = cardActions.find(SyncButton);
        expect(syncButton).toHaveLength(1);
        expect(syncButton.prop('id')).toEqual(space.id);
      });

      it('does not rendes <IconButton /> component', () => {
        expect(wrapper.find(IconButton)).toHaveLength(0);
      });
    });
  });
});
