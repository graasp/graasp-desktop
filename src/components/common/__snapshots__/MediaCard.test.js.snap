// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCard /> <MediaCard /> with showActions = false renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className=""
>
  <WithStyles(ForwardRef(CardActionArea))
    onClick={[MockFunction]}
  >
    <WithStyles(ForwardRef(CardMedia))
      className=""
      image="image"
      title="name"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        variant="h5"
      >
        name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    disableSpacing={true}
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CardContent))
      className=""
    >
      <Text
        className=""
        content="text"
        style={Object {}}
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(ForwardRef(Card))>
`;

exports[`<MediaCard /> <MediaCard /> with showActions = true with text defined renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className=""
>
  <WithStyles(ForwardRef(CardActionArea))
    onClick={[MockFunction]}
  >
    <WithStyles(ForwardRef(CardMedia))
      className=""
      image="image"
      title="name"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        variant="h5"
      >
        name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    disableSpacing={true}
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CardContent))
      className=""
    >
      <Text
        className=""
        content="text"
        style={Object {}}
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Collapse))>
  <WithStyles(ForwardRef(CardActions))
    disableSpacing={true}
  >
    <withI18nextTranslation(WithStyles(undefined))
      id="id"
    />
    <withI18nextTranslation(WithStyles(undefined))
      space={
        Object {
          "id": "id",
          "name": "name",
        }
      }
    />
    <withI18nextTranslation(WithStyles(undefined))
      id="id"
    />
    <WithStyles(ForwardRef(IconButton))
      aria-expanded={false}
      aria-label="Show more"
      className=""
      onClick={[Function]}
    >
      <ExpandMoreIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`<MediaCard /> <MediaCard /> with showActions = true with text undefined renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className=""
>
  <WithStyles(ForwardRef(CardActionArea))
    onClick={[MockFunction]}
  >
    <WithStyles(ForwardRef(CardMedia))
      className=""
      image="image"
      title="name"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        variant="h5"
      >
        name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    disableSpacing={true}
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CardContent))
      className=""
    >
      <Text
        className=""
        content=""
        style={Object {}}
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Collapse))>
  <WithStyles(ForwardRef(CardActions))
    disableSpacing={true}
  >
    <withI18nextTranslation(WithStyles(undefined))
      id="id"
    />
    <withI18nextTranslation(WithStyles(undefined))
      space={
        Object {
          "id": "id",
          "name": "name",
        }
      }
    />
    <withI18nextTranslation(WithStyles(undefined))
      id="id"
    />
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`<MediaCard /> <MediaCard /> with showActions = undefined renders correctly 1`] = `
<WithStyles(ForwardRef(Card))
  className=""
>
  <WithStyles(ForwardRef(CardActionArea))
    onClick={[MockFunction]}
  >
    <WithStyles(ForwardRef(CardMedia))
      className=""
      image="image"
      title="name"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        variant="h5"
      >
        name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(Collapse))
    disableSpacing={true}
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <WithStyles(ForwardRef(CardContent))
      className=""
    >
      <Text
        className=""
        content="text"
        style={Object {}}
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(ForwardRef(Card))>
`;
